code/modules/power/apc.dm

```diff
@@ -1279,11 +1279,35 @@ obj/machinery/power/apc/proc/autoset(var/val, var/on)
 	src.visible_message("<span class='notice'>[src]'s screen flickers with warnings briefly!</span>")
 	spawn(rand(2,5))
 		src.visible_message("<span class='notice'>[src]'s screen suddenly explodes in rain of sparks and small debris!</span>")
+		spark(get_turf(src), 8)
 		stat |= BROKEN
 		operating = 0
 		update_icon()
 		update()
 
+
+//Very violent failure caused by power overload. Dangerous to nearby personnel
+/obj/machinery/power/apc/proc/rupture()
+	var/turf/T = get_turf(src)
+	for (var/mob/living/L in range(6, src))
+		//Hitting people through walls has a 50% chance to fail
+		if (prob(50) || (L in view(6, src)))

```

Comment:
1. Electricity doesn't go through walls.
2. Please use the tesla's proc, even if it is more complex. It's complex for a reason.